### PR TITLE
[FRAME] Extend the functionality of the Frame template.

### PR DESCRIPTION
**Pull request by pwielders**
Sometimes the ordering in a binary serialzed frame is LSB, sometimes it is MSB. Allow the template to
cope with both situations, the default is MSB as that was as it was used sofar.
Also sometimes we only need to work with very small frames <255 bytes. using the the uint16_t for maintaining
location and size is a bit of overkill. Maybe in the future we would like to enlarge the 64Kb boundary to
4Gb, and the variables than need to be uint32_t :-)
Make the variable sizes for creating the buffers a template parameter, again making the default equal to
what it was.

### CODE REVIEW

File: Source/core/Frame.h [hunk 1]
```diff
@@ -303,59 +310,78 @@ namespace Core {
...
             , _data(buffer, length)
         {
             static_assert(BLOCKSIZE == 0, "This method can only be called if you pass a buffer that can not be extended");
         }
-        ~FrameType()
-        {
+        ~FrameType() = default;
+
+        FrameType<BLOCKSIZE, BIG_ENDIAN_ORDERING, SIZE_CONTEXT>& operator=(const FrameType<BLOCKSIZE, BIG_ENDIAN_ORDERING, SIZE_CONTEXT>& rhs) {
+            Size(rhs.Size());
```
Comment (MFransen69) [inline Source/core/Frame.h]: Add check for assigning to self? (behaviour memcpy in that case is undefined)

File: Source/core/Frame.h [hunk 2]
```diff
@@ -582,15 +657,15 @@ namespace Core {
...
         {
             TYPENAME result;
 
             if ((offset + sizeof(TYPENAME)) > _size) {
                 ::memset(&result, 0, sizeof(result));
-            } else
+            }
+            else if (BIG_ENDIAN_ORDERING == true) {
 #ifdef LITTLE_ENDIAN_PLATFORM
-            {
```
Comment (MFransen69) [inline Source/core/Frame.h:593]: Didn't check very thoroughly but wouldn't it be better to put the endian handling in some helper method? Code is now duplicated it seems and an issue would need to be fixed in two places

**Review by MFransen69 — CHANGES_REQUESTED** [review]

File: Source/core/Frame.h (hunk 2 above)
Comment (pwielders) [reply Source/core/Frame.h:593]: That was quicker to just copy/paste ;-) But created endian handling in separate methods,

File: Source/core/Frame.h (hunk 1 above)
Comment (pwielders) [reply Source/core/Frame.h]: added.

**Review by MFransen69 — APPROVED** [review]

**Pull request merged**